pkg/chunkenc/memchunk.go

```diff
@@ -152,6 +152,94 @@ func (hb *headBlock) serialise(pool WriterPool) ([]byte, error) {
 	return outBuf.Bytes(), nil
 }
 
+// CheckpointBytes is used for WAL checkpointing
+func (hb *headBlock) CheckpointBytes(version byte) ([]byte, error) {
+	// wB is eventually returned via buf.Bytes(), don't return it to the pool.
+	wB := BytesBufferPool.Get(1 << 10).([]byte)
```

Comment:
I think you should just not use the pool here and allocate.

That can give more work to the pool when it needs to sweep ref see if they are in-use.